### PR TITLE
Feature/40314 feat: add Controller/search function

### PR DESCRIPTION
**Pull request by suraj-aggarwal**
[issue no.](https://redmine.successive.tech/issues/40314)

add search function in controller add search in Versionable.

### CODE REVIEW

File: src/controllers/trainee/Controller.ts [hunk 1]
```diff
@@ -1,71 +1,121 @@
...
     }
+  };
+
+  list = async (req: IRequest, res: Response) => {
+    try {
+      const { skip, limit, sort, search, ...query } = req.query;
+      const options: object = { skip, limit, sort };
+      const filter: object = queryString.parse(search);
+      let result;
+      if (Object.keys(filter).length) {
```
Comment (madhavBansal-successiveTech) [inline src/controllers/trainee/Controller.ts:47]: use helper function

File: src/controllers/user/Controller.ts [hunk 2]
```diff
@@ -21,133 +20,53 @@ class UserController {
...
-    } else {
-      res.send(`Invalid email`);
-    }
-  };
   userProfile = (req: Request, res: Response): void => {
     const token: string = req.headers.authorization;
     const decodedPayload: any = jwt.verify(token, configuration.SECRET_KEY);
-    const {id} = decodedPayload;
+    const { id } = decodedPayload;
     console.log(decodedPayload);
```
Comment (madhavBansal-successiveTech) [inline src/controllers/user/Controller.ts]: remove console

File: src/controllers/user/Controller.ts [hunk 3]
```diff
@@ -21,133 +20,53 @@ class UserController {
...
     this.userRepo
       .profile(id)
       .then(profile => {
```
Comment (madhavBansal-successiveTech) [inline src/controllers/user/Controller.ts]: use async await

File: src/repositories/user/UserRepository.ts [hunk 4]
```diff
@@ -4,46 +4,54 @@ import IUserModel from './IUserModel';
...
+  };
+
+  public update = record => {
+    return super.update(record);
+  };
+
+  public isExists = (id: string, email: string) => {
+    console.log('----------isExits-----------', id, email);
+    const condition = { originalId: id, email, deleatedAt: undefined };
+    return userModel.exists(condition);
```
Comment (madhavBansal-successiveTech) [inline src/repositories/user/UserRepository.ts]: should be in versionable

File: src/controllers/user/Controller.ts (hunk 2 above)
Comment (suraj-aggarwal) [reply src/controllers/user/Controller.ts]: Done

File: src/repositories/user/UserRepository.ts (hunk 4 above)
Comment (suraj-aggarwal) [reply src/repositories/user/UserRepository.ts]: Done

**Pull request merged**